Source/Core/Resource.js

```diff
@@ -309,6 +354,54 @@ define([
             get: function() {
                 return isDataUri(this._url);
             }
+        },
+
+        /**
+         * True if the Resource refers to a blob URI.
+         *
+         * @memberof Resource.prototype
+         * @type {Boolean}
+         */
+        isBlobUri: {
+            get: function() {
+                return isBlobUri(this._url);
+            }
+        },
+
+        /**
+         * True if the Resource refers to a cross origin URL.
+         *
+         * @memberof Resource.prototype
+         * @type {Boolean}
+         */
+        isCrossOriginUrl: {
+            get: function() {
+                return isCrossOriginUrl(this._url);
+            }
+        },
+
+        /**
+         * True if the Resource has request headers. This is equivalent to checking if the headers property has any keys.
+         *
+         * @memberof Resource.prototype
+         * @type {Boolean}
+         */
+        hasHeaders: {
+            get: function() {
+                return (Object.keys(this.headers).length > 0);
+            }
+        },
+
+        /**
+         * Returns the last response received when this resource was requested. Contains status and headers properties. Header keys are all lower case.
+         *
+         * @memberof Resource.prototype
+         * @type {Object}
+         */
+        lastResponse: {
```

Comment:
Did this property exist elsewhere already, or is it new?  Are we worried it could lead to bloated memory usage?

Reply:
The idea was to set the response that is returned so the caller can get information about the response besides just the data. This would make it possible to avoid hacks like https://github.com/AnalyticalGraphicsInc/cesium/issues/1353. It's not used so I can remove it.

All imagery and terrain providers discard the resource once its received, so we'd probably be ok as far as bloating goes.